test/functional/collations_tests.js

```diff
@@ -668,7 +668,6 @@ describe('Collation', function() {
             .collection('test')
             .findOne({ a: 1 }, { collation: { caseLevel: true } }, function(err) {
               test.equal(`server ${singleServer.uri()} does not support collation`, err.message);
-              singleServer.destroy();
```

Comment:
Before taking this line out, the test case itself passed, but the `afterEach` hook failed with the following error: 
```
  1) Collation "after each" hook for "Fail due to no support for collation":
     Error: Not running
      at Server.close (net.js:1608:12)
      at emitCloseNT (net.js:1659:8)
      at _combinedTickCallback (internal/process/next_tick.js:135:11)
      at Immediate._tickCallback (internal/process/next_tick.js:180:9)
```
The afterEach hook is just `afterEach(() => mock.cleanup())`, so it seems like its erroring because it can't cleanup something that was already destroyed. Not sure why this error wasn't happening previously.

Reply:
Is it possible that this is because we're `throw`ing the error about collation, rather than calling back with a `MongoError` instance?